druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java

```diff
@@ -875,7 +875,14 @@ protected JsonAggregation getJsonAggregation(List<String> fieldNames,
               + " because an approximate count distinct is not acceptable.");
         }
       }
-      aggregation = new JsonAggregation("count", name, only);
+      if (aggCall.getArgList().size() == 1) {
+        final JsonFilter filter = new JsonCompositeFilter(JsonFilter.Type.NOT,
```

Comment:
@b-slim , can we add a comment here explaining what this is doing? It does not seem straightforward unless you understand well how Druid works (specially the ```JsonSelector(only, null, null)```)

Reply:
done